lib/craft/clock_bound/client.ex

```diff
@@ -0,0 +1,217 @@
+defmodule Craft.ClockBound.Client do
+  @moduledoc """
+  This module provides functionality to retrieve clock error bounds and clock status from shared memory.
+
+  ## Features
+
+  - Reads binary data from a shared memory file and parses it according to the ClockBound protocol specification.
+  - Provides the current clock error bounds (`earliest` and `latest`) and clock status.
+  - Converts timestamps to `DateTime` for easier comparision.
+
+  ## Usage
+
+  1. Call `ClockBound.Client.now/0` or `ClockBound.Client.now/1` to retrieve the current clock error bounds and status:
+    ```elixir
+    case ClockBound.Client.now() do
+      {:ok, clock_data} ->
+        IO.inspect(clock_data)
+
+      {:error, reason} ->
+        _
+    end
+    ```
+  2. Call `ClockBound.Client.compare/2` to compare two ClockBound windows:
+    ```elixir
+    case ClockBound.Client.compare(clock_data_1, clock_data_2) do
+      :lt -> IO.puts("Clock 1 is less than Clock 2")
+      :gt -> IO.puts("Clock 1 is greater than Clock 2")
+      :ov -> IO.puts("Clock 1 and Clock 2 overlap")
+    end
+    ```
+
+  ## Notes
+
+  - The default shared memory path is `/var/run/clockbound/shm`. You can override this by passing a custom path to the functions.
+  - The module adheres to the ClockBound protocol specification, which can be found [here](https://github.com/aws/clock-bound/blob/main/docs/PROTOCOL.md).
+  """
+
+  alias Craft.ClockBound.Client
+  alias Craft.ClockBound.Native
+
+  @default_shm_path Application.compile_env(:clock_bound, :shm_path)
```

Comment:
Double check with @koudelka on what he had in mind, but I think it would make sense to have the default here instead of the config.
I believe the config set in a mix project only effects that project, so this config value will not work as a default when used from within authz. It would need to get explicitly set within the authz config.
```suggestion
  @default_shm_path Application.compile_env(:clock_bound, :shm_path, "/var/run/clockbound/shm")
```
This way, we set the default here in code, but authz can override this value from it's config if it wants.

Side note, I would probably also namespace this config within `:craft` not top level :clock_bound since there might later be an official library for clock bound that uses this config.

Reply:
yes to all of that, thanks. :)